### ISSUE TITLE
training accuracy different from history object

### ISSUE THREAD

**Issue by ggous** [tensorflow/tensorflow#51541]
Hi , I am dealing with a problem on the accuracy history.
As you can see, during training, we have:

```
Epoch 1/10
597/597 [==============================] - 4s 3ms/step - loss: 0.6822 - accuracy: 0.7007 - val_loss: 0.6607 - val_accuracy: 0.9228
Epoch 2/10
597/597 [==============================] - 1s 2ms/step - loss: 0.6675 - accuracy: 0.7838 - val_loss: 0.6463 - val_accuracy: 0.9228
Epoch 3/10
597/597 [==============================] - 1s 2ms/step - loss: 0.6537 - accuracy: 0.8284 - val_loss: 0.6329 - val_accuracy: 0.9228
Epoch 4/10
597/597 [==============================] - 2s 3ms/step - loss: 0.6408 - accuracy: 0.8598 - val_loss: 0.6204 - val_accuracy: 0.9228
Epoch 5/10
597/597 [==============================] - 1s 2ms/step - loss: 0.6287 - accuracy: 0.8876 - val_loss: 0.6087 - val_accuracy: 0.9228
Epoch 6/10
597/597 [==============================] - 1s 2ms/step - loss: 0.6174 - accuracy: 0.9024 - val_loss: 0.5978 - val_accuracy: 0.9228
Epoch 7/10
597/597 [==============================] - 1s 2ms/step - loss: 0.6068 - accuracy: 0.9103 - val_loss: 0.5875 - val_accuracy: 0.9228
Epoch 8/10
597/597 [==============================] - 1s 2ms/step - loss: 0.5968 - accuracy: 0.9146 - val_loss: 0.5779 - val_accuracy: 0.9228
Epoch 9/10
597/597 [==============================] - 1s 2ms/step - loss: 0.5874 - accuracy: 0.9146 - val_loss: 0.5688 - val_accuracy: 0.9228
Epoch 10/10
597/597 [==============================] - 1s 2ms/step - loss: 0.5785 - accuracy: 0.9146 - val_loss: 0.5603 - val_accuracy: 0.9228
```

So, accuracy reaches 0.9146.

But, If you see the accuracy values in the history:

```
history.history['accuracy']

[0.7472780346870422,
 0.7945979833602905,
 0.8073701858520508,
 0.8146985173225403,
 0.82097989320755,
 0.8243299722671509,
 0.8255862593650818,
 0.8262144327163696,
 0.8262144327163696,
 0.8262144327163696]
```

are different!

Code:

```
import pandas as pd
import numpy as np
import tensorflow as tf
import matplotlib.pyplot as plt
from tensorflow.keras.layers import Dense, LSTM, \
    Bidirectional, RepeatVector
from tensorflow.keras import Sequential

X_train = np.load('./X_train.npy')
X_val = np.load('./X_val.npy')
y_train = np.load('./y_train.npy')
y_val = np.load('./y_val.npy')


model = Sequential()
model.add(Bidirectional(LSTM(2,
                        return_sequences=False,
                        activation='tanh',
                    input_shape=(12,
                                 10))))
model.add(RepeatVector(8))
# Decoder
model.add((LSTM(4,
                activation='tanh',
                return_sequences=True)))

model.add((Dense(1, activation='sigmoid')))
                    
model.compile(optimizer=tf.keras.optimizers.SGD(learning_rate=0.0001), 
              loss='binary_crossentropy',
              metrics=['accuracy'])

epochs = 10
batch_size = 1

history = model.fit(
    X_train,
    y_train, 
    validation_data = (X_val, y_val),
    steps_per_epoch=int(len(X_train) / batch_size),
    validation_steps=int(len(X_val) / batch_size),
    epochs=epochs,
    batch_size=batch_size,
    shuffle=False)


fig, axes = plt.subplots(figsize=(20, 12))
axes.plot(history.epoch, history.history['accuracy'], label = 'Train acc')
axes.plot(history.epoch, history.history['val_accuracy'], label = 'Val acc')
axes.legend()
```

You can download data [here](https://easyupload.io/m/5iv09z)

<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
You can see it at the end

**Describe the current behavior**

 Different training accuracy results between history object and history when printing on screen

**Describe the expected behavior**
 We should have the same results!

**Standalone code to reproduce the issue**
At the beginning

                  **System Info**
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------

== check python ===================================================
python version: 3.8.10
python branch: 
python build version: ('default', 'Jun  2 2021 10:49:15')
python compiler version: GCC 9.4.0
python implementation: CPython

== check os platform ===============================================

== are we in docker =============================================
No

== compiler =====================================================
c++ (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

== check pips ===================================================
protobuf                3.6.1               

== check for virtualenv =========================================
False

== env ==========================================================
LD_LIBRARY_PATH is unset
DYLD_LIBRARY_PATH is unset

== nvidia-smi ===================================================
Wed Aug 18 09:48:39 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.91.03    Driver Version: 460.91.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce GTX 1650    On   | 00000000:01:00.0  On |                  N/A |
| 24%   37C    P8    10W /  90W |   1096MiB /  3907MiB |      9%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1085      G   /usr/lib/xorg/Xorg                275MiB |
|    0   N/A  N/A      1727      G   cinnamon                           74MiB |
|    0   N/A  N/A      3228      G   ...AAAAAAAAA= --shared-files       30MiB |
|    0   N/A  N/A      3715      G   /usr/lib/firefox/firefox          125MiB |
|    0   N/A  N/A      3834      G   /usr/lib/firefox/firefox            1MiB |
|    0   N/A  N/A      4004      G   /usr/lib/firefox/firefox            1MiB |
|    0   N/A  N/A      4518      G   ...onda3/envs/dpl/bin/python        1MiB |
|    0   N/A  N/A      4593      C   ...onda3/envs/dpl/bin/python      577MiB |
+-----------------------------------------------------------------------------+

== cuda libs  ===================================================

== tensorflow installed from info ==================

== python version  ==============================================
(major, minor, micro, releaselevel, serial)
(3, 8, 10, 'final', 0)

== bazel version  ===============================================

== check python ===================================================
python version: 3.8.8
python branch: 
python build version: ('default', 'Feb 24 2021 21:46:12')
python compiler version: GCC 7.3.0
python implementation: CPython

== check os platform ===============================================

== are we in docker =============================================
No

== compiler =====================================================
c++ (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

== check pips ===================================================
neptune-tensorflow-keras          0.9.1
numpy                             1.19.2
numpydoc                          1.1.0
protobuf                          3.14.0
tensorflow                        2.4.1
tensorflow-datasets               1.2.0
tensorflow-estimator              2.4.0
tensorflow-metadata               0.14.0

== check for virtualenv =========================================
False

== tensorflow import ============================================
tf.version.VERSION = 2.4.1
tf.version.GIT_VERSION = unknown
tf.version.COMPILER_VERSION = 5.4.0
      8883:	find library=libpthread.so.0 [0]; searching

== nvidia-smi ===================================================
Wed Aug 18 09:49:20 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.91.03    Driver Version: 460.91.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce GTX 1650    On   | 00000000:01:00.0  On |                  N/A |
| 24%   36C    P8    10W /  90W |   1107MiB /  3907MiB |      6%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1085      G   /usr/lib/xorg/Xorg                284MiB |
|    0   N/A  N/A      1727      G   cinnamon                           77MiB |
|    0   N/A  N/A      3228      G   ...AAAAAAAAA= --shared-files       30MiB |
|    0   N/A  N/A      3715      G   /usr/lib/firefox/firefox          125MiB |
|    0   N/A  N/A      3834      G   /usr/lib/firefox/firefox            1MiB |
|    0   N/A  N/A      4004      G   /usr/lib/firefox/firefox            1MiB |
|    0   N/A  N/A      4518      G   ...onda3/envs/dpl/bin/python        1MiB |
|    0   N/A  N/A      4593      C   ...onda3/envs/dpl/bin/python      577MiB |
+-----------------------------------------------------------------------------+

== cuda libs  ===================================================

== tensorflow installed from info ==================
Name: tensorflow
Version: 2.4.1
Summary: TensorFlow is an open source machine learning framework for everyone.
Home-page: https://www.tensorflow.org/
Author-email: packages@tensorflow.org
License: Apache 2.0
Location: /home/ggousios/miniconda3/envs/dpl/lib/python3.8/site-packages
Required-by: neptune-tensorflow-keras, neptune-tensorboard

== python version  ==============================================
(major, minor, micro, releaselevel, serial)
(3, 8, 8, 'final', 0)

== bazel version  ===============================================

**Comment by sushreebarsa** [comment]
@ggous In order to expedite the trouble-shooting process,Can you please fill the issue [template](https://github.com/tensorflow/tensorflow/issues/new/choose).Thanks!

**Comment by ggous** [comment]
I have updated. I am not sure if this is exactly what you want.
I used the template at the end of comment.

**Comment by Abhishek-TyRnT** [comment]
Just update to tensorflow 2.6......Don't see the same error over there

**Comment by sushreebarsa** [comment]
@ggous It looks like you are using an older Version of Tensorflow (2.4.1). Could you please execute your code using Latest stable Version of TF (2.6.0) and let us know if the issue still persists? Please refer to the [link](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/SGD) & let us know if it helps?Thanks!

**Comment by ggous** [comment]
Yes! It works fine with tf 2.6.0 ! 
Thanks!
